key.go

```diff
@@ -0,0 +1,778 @@
+package cose
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"errors"
+	"fmt"
+	"math/big"
+	"strconv"
+
+	cbor "github.com/fxamacker/cbor/v2"
+)
+
+const (
+	// The key is used to create signatures. Requires private key fields.
+	KeyOpSign KeyOp = 1
+
+	// The key is used for verification of signatures.
+	KeyOpVerify KeyOp = 2
+
+	// the key is used for key transport encryption.
+	KeyOpEncrypt KeyOp = 3
+
+	// the key is used for key transport decryption. Requires private key fields.
+	KeyOpDecrypt KeyOp = 4
+
+	// The key is used for key wrap encryption.
+	KeyOpWrapKey KeyOp = 5
+
+	// The key is used for key wrap decryption.
+	KeyOpUnwrapKey KeyOp = 6
+
+	// The key is used for deriving keys. Requires private key fields.
+	KeyOpDeriveKey KeyOp = 7
+
+	// The key is used for deriving bits not to be used as a key. Requires
+	// private key fields.
+	KeyOpDeriveBits KeyOp = 8
+
+	// The key is used for creating MACs.
+	KeyOpMacCreate KeyOp = 9
+
+	// The key is used for validating MACs.
+	KeyOpMacVerify KeyOp = 10
+)
+
+// KeyOp represents a key_ops value used to restrict purposes for which a Key
+// may be used.
+type KeyOp int64
+
+// KeyOpFromString returns the KeyOp corresponding to the specified name.
+func KeyOpFromString(val string) (KeyOp, error) {
```

Comment:
a similar comment I made for `AlgorithmFromString` (i.e., likely superfluous `error` returned)